plugins/HHAnalyzer.cc

```diff
@@ -600,6 +600,21 @@ void HHAnalyzer::analyze(const edm::Event& event, const edm::EventSetup&, const
             myllmetjj.melaAngles = getMELAAngles(llmet[illmet].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4);
             myllmetjj.visMelaAngles = getMELAAngles(ll[ill].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4); // only take the visible part of the H(ww) candidate
 
+            // Compute MT2. See https://arxiv.org/pdf/1309.6318v1.pdf and https://arxiv.org/pdf/1411.4312v5.pdf
+            constexpr double M_B = 4.66;
+            constexpr double M_W = 80.385;
+            double px_invisible = myllmetjj.lep1_p4.px() + myllmetjj.lep2_p4.px() + myllmetjj.met_p4.px();
+            double py_invisible = myllmetjj.lep1_p4.py() + myllmetjj.lep2_p4.py() + myllmetjj.met_p4.py();
+
+            myllmetjj.MT2 = asymm_mt2_lester_bisect::get_mT2(
+                    M_B, myllmetjj.jet1_p4.px(), myllmetjj.jet1_p4.py(),
+                    M_B, myllmetjj.jet2_p4.px(), myllmetjj.jet2_p4.py(),
+                    px_invisible, py_invisible,
+                    M_W, M_W,
+                    1 // Absolute precision
+                    );
```

Comment:
after some discussion with @l-cadamuro about [the H(bb)H(tautau) implementation](https://github.com/LLRCMS/KLUBAnalysis/blob/457d6604761b7ce47f9621c356cffda7d636d11b/test/skimNtuple.cpp#L2035-L2057), we have quite a couple of things wrong (also my own reading :smile:)
- the H(bb)H(tautau) follows the implementation from [1411.4312](https://arxiv.org/pdf/1411.4312v5.pdf) which has different conventions than [1309.6318](https://arxiv.org/pdf/1309.6318v1.pdf), in particular wrt what is called 'visible' and 'invisible' so indeed your interpretation is more correct than mine in that respect
- however, the idea is to have all inputs purely from data and no external input such as PDG masses, in the end this is at heart a kinematic fit, which will happen to be bounded for some physical processes _by construction_, and not _by constraining masses to match a target_, though this I agree is not clearly stated in either paper as far as I could spot...

bref, the fixes needed would be:
- the first (second) `M_B` would be `myllmetjj.jet1(2)_p4.M()`
- the first (second) `M_W` would need to be `myllmetjj.lep1(2)_p4.M()`
- if the absolute precision means the precision on `MT2` itself, probably we could move to `0.5` in order to consider cuts at integer values? This may avoid numerical instabilities (or bin migration)?

Note to self:
- yes, it is possible to have this variable computed for each leg of the di-higgs system hypothesis (instead of the di-top system hypothesis), but performance should be similar and you probably are anyway more concerned about a sharp behaviour to cut away ttbar